src/components/RegisterForm/RegisterForm.jsx

```diff
@@ -42,19 +47,28 @@ export default function RegisterForm() {
 
   function onSubmit(event) {
     event.preventDefault();
-    dispatch(adminRegister(values.email, values.password, values.company)).then(
-      () => setOpenReg(false)
-    );
+    validate();
+    if (isValid) {
+      dispatch(adminRegister(values.email, values.password, values.company)).then(
```

Comment:
```suggestion
      dispatch(adminRegister(values)).then(
```
Лучше пусть эта функция принимает объект. Все эти вещи связаны между собой. Если кто-то захочет прислать в нее один емейл, а пароль от другого емейла, вот это будет нонсенс. Будет хорошо, если код не будет давать возможности наступить на грабли, и логика в коде будет максимально приближена к бизнес логике